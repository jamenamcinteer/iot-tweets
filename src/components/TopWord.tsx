import React from "react"
import styled from "styled-components"
import { IPropsTheme, ITopWordProps } from "../interfaces/interfaces"

const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  pointer-events: none;
`

const Word = styled.p`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeLarge};
  font-weight: bold;
  margin: 0;
  margin-bottom: .5rem;
  padding: 0;
`
const Count = styled.p`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeSmall};
  margin: 0;
  padding: 0;
`

const TopWord = ({ topWord }: ITopWordProps) => {
  return (
    <Container>
      {topWord && topWord.word && (
        <div>
          <Word data-testid="top-word">{topWord.word}</Word>
          <Count data-testid="top-wordcount">used {topWord.count} times</Count>
        </div>
      )}
    </Container>
  )
}

export default TopWord