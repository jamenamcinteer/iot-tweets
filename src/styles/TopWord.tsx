import styled from "styled-components"
import { IPropsTheme } from "../interfaces/interfaces"

export const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  pointer-events: none;
`

export const Word = styled.p`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeLarge};
  font-weight: bold;
  margin: 0;
  margin-bottom: .5rem;
  padding: 0;
`
export const Count = styled.p`
  font-size: ${(props: IPropsTheme) => props.theme.fontSizeSmall};
  margin: 0;
  padding: 0;
`