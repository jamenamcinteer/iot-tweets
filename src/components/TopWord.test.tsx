import React from 'react'
import { cleanup, render } from "@testing-library/react"
import "jest-styled-components"
import TopWord from './TopWord'

afterEach(cleanup)

it('renders as expected', () => {
  const topWord = {
    word: "AI",
    count: 50,
    index: 0
  }

  const { asFragment } = render(<TopWord topWord={topWord} />)
  expect(asFragment()).toMatchSnapshot()
})