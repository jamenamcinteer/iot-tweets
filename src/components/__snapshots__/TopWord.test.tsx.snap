// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
}

.c1 {
  font-size: 1.8rem;
  font-weight: bold;
  margin: 0;
  margin-bottom: .5rem;
  padding: 0;
}

.c2 {
  font-size: 1rem;
  margin: 0;
  padding: 0;
}

<div
    class="c0"
  >
    <div>
      <p
        class="c1"
        data-testid="top-word"
      >
        AI
      </p>
      <p
        class="c2"
        data-testid="top-wordcount"
      >
        used 50 times
      </p>
    </div>
  </div>
</DocumentFragment>
`;
